Legacy/bonej/src/main/java/org/bonej/plugins/ParticleDisplay.java

```diff
@@ -450,13 +499,19 @@ static Color3f colourFromEigenVector(EigenvalueDecomposition eigen) {
 
 		final int rgb = Color.HSBtoRGB(hue, saturation, brightness);
 		final Color color = new Color(rgb);
-		float red = (float) (color.getRed() / 255d);
-		float green = (float) (color.getGreen() / 255d);
-		float blue = (float) (color.getBlue() / 255d);
+		float red = color.getRed() / 255f;
+		float green = color.getGreen() / 255f;
+		float blue = color.getBlue() / 255f;
 
 		return new Color3f(red, green, blue);
 	}
 
+	/**
+	 * Check whether ellipsoid radii are NaN
+	 * 
+	 * @param radii list of radii
+	 * @return true if all radii are not NaN
+	 */
 	private static boolean isRadiiValid(final double[] radii) {
```

Comment:
I noticed that in `displayEllipsoids` after you call `isRadiiValid` you check that `a < b < c`. That could be performed inside the method as well.

Reply:
This error is never logged in practice, I suspect because the eigendecomposition sorts the eigenvalues and correspondingly the radii.